packages/pyinstrument/test_pyinstrument.py

```diff
@@ -0,0 +1,20 @@
+import pytest
+from pytest_pyodide import run_in_pyodide
+
+
+@pytest.mark.skip_refcount_check
+@run_in_pyodide(packages=["pyinstrument"])
+async def test_pyinstrument(selenium):
+    """Check that we can run the profiler on async code
+
+    without errors.
+    """
+    import asyncio
+
+    from pyinstrument import Profiler
+
+    p = Profiler()
+    with p:
+        await asyncio.sleep(0.1)
+
+    p.print()
```

Comment:
Maybe assert that p.output_text() matches some regex?

Reply:
There's also a JSON output available from `p.last_session.save('foo.pyisession')`, which might be more palatable to parse and less likely to change for purely cosmetic reasons... is there a way to play around with it on a CI output somewhere? I must confess I haven't gotten my local dev setup working in a while.

Reply:
You can download the build artifacts from: https://output.circle-artifacts.com/output/job/7ffb988b-80ed-42ec-8d3f-debebcda4dc3/artifacts/0/root/repo/pyodide.tar.gz

Reply:
here's a thing... i feel like the JSON output is the artifact that would actually be saved as part of a benchmark, so having a handle on what can be known about it seems important, even if you'd generally reload the output. It does seem a bit odd that the JSON output doesn't break up the raw record data more (just strings with ansi codes) but meh.

```py
import asyncio
import json
from pathlib import Path

from pyinstrument.profiler import Profiler, Session

with Profiler() as p1:
    await asyncio.sleep(0.1)

p1.print(show_all=True, timeline=True)

session_file = Path("foo.pyisession")
s1 = p1.last_session
s1.save(session_file)

json_opts = dict(indent=2, sort_keys=True)
s1_data = json.load(session_file.open())
s1_json = json.dumps(s1_data, **json_opts)

# _all_ the keys as of 4.4.0, but presumably more could be added
expected_keys = {
    "cpu_time",
    "duration", 
    "frame_records", 
    "program", 
    "sample_count", 
    "start_call_stack", 
    "start_time", 
}
missing_keys = expected_keys - set(s1_data)

assert not missing_keys, f"session JSON missing: {missing_keys}"

s2 = Session.load(session_file)
s2_json = json.dumps(s2.to_json(), **json_opts)

assert s1_json == s2_json, "loaded JSON did not match"
```

Reply:
Thanks, @bollwyvl, added your extensive test.